packages/app/src/components/ModeSelector.vue

```diff
@@ -0,0 +1,144 @@
+          color="primary"
+          icon="help_outline"
+        />
+        <q-btn v-else @click="setQuestionsMode" round size="40px" icon="help_outline" />
+        <p style="text-align:center;">How many question would you like to play?</p>
+      </div>
+    </div>
+    <div class="conatiner" v-if="isTimedMode">
+      <p>How long would you like to practice for?</p>
+      <q-select
+        filled
+        v-model="selectedTime"
+        :options="timeOptions"
+        label="Choose Time"
+        style="width: 250px"
+      />
+    </div>
+    <div class="conatiner" v-if="isQuestionsMode">
+      <p>How many questions would you like to practice</p>
+      <q-select
+        filled
+        v-model="selectedQuestionCount"
+        :options="questionOptions"
+        label="Choose Question"
+        style="width: 250px"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { mapGetters, mapActions } from "vuex";
+import { PracticeMode } from "../engine/models/math_question";
+import { PracticeActions, PracticeGetters } from "../store/practice/practice";
+
+export default {
+  data() {
+    return {
+      selectedQuestionCount: 10,
+      selectedTime: 1,
+      timeOptions: [
+        { label: "1 minute", value: 1 },
+        { label: "3 minutes", value: 3 },
+        { label: "5 minutes", value: 5 },
+        { label: "8 minutes", value: 8 },
+        { label: "10 minutes", value: 10 }
+      ],
+      questionOptions: [
+        { label: "10 minute", value: 10 },
+        { label: "20 minutes", value: 20 },
+        { label: "25 minutes", value: 25 },
+        { label: "50 minutes", value: 50 },
+        { label: "100 minutes", value: 100 }
+      ]
+    };
+  },
+  mounted() {
+    const app: any = this;
+    app.selectedTime = Math.round(app.practiceTime / 60);
+    app.selectedQuestionCount = app.practiceQuestionCount;
+  },
+  methods: {
+    setTimedMode() {
+      this.setPracticeMode(PracticeMode.TIME);
+    },
+    setQuestionsMode() {
+      this.setPracticeMode(PracticeMode.QUESTIONS);
+    },
+    ...mapActions({
+      setPracticeQuestionCount: PracticeActions.SET_PRACTICE_QUESTION_COUNT,
+      setPracticeMode: PracticeActions.SET_PRACTICE_MODE,
+      setPracticeTime: PracticeActions.SET_PRACTICE_TIME
+    })
+  },
+  computed: {
+    isQuestionsMode() {
+      const app: any = this;
+      return app.practiceMode === PracticeMode.QUESTIONS;
+    },
+    isTimedMode() {
+      const app: any = this;
+      return app.practiceMode === PracticeMode.TIME;
+    },
+    ...mapGetters({
+      practiceMode: PracticeGetters.PRACTICE_MODE,
+      practiceQuestionCount: PracticeGetters.PRACTICE_QUESTION_COUNT,
+      practiceTime: PracticeGetters.PRACTICE_TIME
+    })
+  },
+  watch: {
+    selectedQuestionCount(newValue: any) {
+      const app: any = this;
+      if (typeof newValue.value !== "undefined")
+        app.setPracticeQuestionCount(newValue.value);
+      else app.setPracticeQuestionCount(newValue);
```

Comment:
I don't understand this if statement. In what scenario does the else case happen?

Reply:
It is possible for the QuestionCount to be undefined. I am not sure when and why I say the QuestionCount to be undefined. However, once I saw the QuestionCount be undefined.